src/routes/trips.js

```diff
@@ -324,20 +324,188 @@ import validateOneWayTrip from '../middlewares/validateOneWay';
+           "summary": "Search Functionality",
+           "tags": [
+             "Trips"
+           ],
+           "operationId": "SearchFunctionality",
+           "deprecated": false,
+           "produces": [
+             "application/json"
+           ],
+           "parameters": [
+                       {
+               "name": "targetKey",
+               "in": "path",
+               "required": true,
+               "type": "string",
+               "description": "your desired search"
+             },
+                                 {
+               "name": "filterKey",
+               "in": "path",
+               "required": true,
+               "type": "string",
+               "description": "target key"
+             },
+           ],
+           "responses": {
+             "200": {
+               "description": "",
+               "headers": {}
+             }
+           }
+         }
+       }
+    */
 
+/**
+    * @swagger
+    *   "/trips/search?{targetKey}={filterKey}": {
+         "get": {
+           "description": "As a user, I should be able to use the search component\nSo that, I can easily retrieve records from both the request and approval table",
+           "summary": "Search Functionality",
+           "tags": [
+             "Trips"
+           ],
+           "operationId": "SearchFunctionality",
+           "deprecated": false,
+           "produces": [
+             "application/json"
+           ],
+           "parameters": [
+                       {
+               "name": "targetKey",
+               "in": "path",
+               "required": true,
+               "type": "string",
+               "description": "your desired search"
+             },
+                                 {
+               "name": "filterKey",
+               "in": "path",
+               "required": true,
+               "type": "string",
+               "description": "target key"
+             },
+           ],
+           "responses": {
+             "200": {
+               "description": "",
+               "headers": {}
+             }
+           }
+         }
+       }
+    */
+/**
+ * @swagger
+ *  "/trips/edit/": {
+      "post": {
+        "description": "A user can edit an open request.",
+        "summary": "Edit a request.",
+        "tags": [
+          "Trips"
+        ],
+        "deprecated": false,
+        "produces": [
+          "application/json"
+        ],
+        "consumes": [
+          "application/x-www-form-urlencoded"
+        ],
+        "parameters": [
+          {
+              "name": "requestId",
+              "in": "query",
+              "description": "ID of a request to be updated",
+              "required": true,
+              "type": "integer",
+              "format": "int64"
+          },
+          {
+            "name": "origin",
+            "in": "formData",
+            "required": true,
```

Comment:
when you are doing edit you do not force the field to be require you edit fields that you want

Reply:
True, I will have to edit that!